_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
 
-## Test Server-side JavaScript with XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $count := 3
+let $testSequence := (1,2)
+let $errorMessage := "Assert All Exist failed"
+let $result := 
+try{
+  test:assert-all-exist($count, $testSequence)
+} catch($err) {
+  $err
+}
+return test:assert-equal($result//error:code/string(), $errorMessage)
+```
+### Scenario 3 - Test assert same values with same sequences for left and right
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $leftSequence := ("a","b","c")
+let $rightSequence := ("a","b","c")
+return test:assert-same-values($leftSequence, $rightSequence)
+```
+### Scenario 4 - Test assert same values with different sequences for left and right. This generates an exception that is validated with the actual value to get success status
+```
+ xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $leftSequence := ("a","d","c")
+let $rightSequence := ("a","b","c")
+let $formattedErrorMessage := "Assert Equal failed (ASSERT-EQUAL-FAILED): a c d a b c"
+return 
+try{
+  test:assert-same-values($leftSequence, $rightSequence)
+} catch($err) {
+  test:assert-equal($err//error:format-string/string(), $formattedErrorMessage)
+}
+```
+### Scenario 5 - Test assert equal xml - all attributes and elements matched
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ $err
+}
+```
+### Scenario 6 - Test assert equal xml - attributes mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="expected" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): actual expected  :  mismatched attribute text ($expected= /t:test/@type , $actual= /t:test/@type )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
```

Comment:
Let's simplify this example so we're not trying to assert against the failure.  Just note that this is a failing test.

```
let $expected := <xml>expected value</xml>
let $actual := <xml>unexpected value</xml>
return test:assert-equal-xml($expected, $actual)
```
Note that we should define the expected before the actual since that's the pattern used in the assertion method signature.